library/src/main/java/com/getstream/sdk/chat/Chat.kt

```diff
@@ -0,0 +1,76 @@
+package com.getstream.sdk.chat
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import com.getstream.sdk.chat.UrlSigner.DefaultUrlSigner
+import com.getstream.sdk.chat.enums.OnlineStatus
+import com.getstream.sdk.chat.navigation.ChatNavigationHandler
+import com.getstream.sdk.chat.navigation.ChatNavigator
+import com.getstream.sdk.chat.style.ChatFonts
+import com.getstream.sdk.chat.style.ChatFontsImpl
+import com.getstream.sdk.chat.style.ChatStyle
+import com.getstream.sdk.chat.utils.strings.ChatStrings
+import com.getstream.sdk.chat.utils.strings.ChatStringsImpl
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.socket.InitConnectionListener
+import io.getstream.chat.android.livedata.ChatDomain
+import java.lang.IllegalStateException
+
+interface Chat {
```

Comment:
Do we really need this interface?
`ChatImpl` is a class that only implements this interface, doesn't hide anything else. This interface doesn't have any "boundary" reason neither it is implemented in different classes. Maybe we can simplify it

Reply:
Yes, I agree this would be a good idea to refactor this further and get rid of the interface because of the reasons you mentioned

Reply:
Interface might be useful for customers who want to mock/fake chat for testing or other purposes. It might help also to migrate them to another provider if needed. 
And as a user of a library I find it much easier to read interface rather than stuffed with code implementation.

Reply:
A class can be mock/fake without an interface, it is not extremely needed.
If we have our public method documented with some Javadoc, and it is shown by the IDE, the user doesn't need to go to the code implementation.
On the other hand, if a user wants to go to the source code, it is because he wants to know what it is doing under the wood. Having an interface that hides it is an extra step the user needs to do

Reply:
I'd park this topic atm.
This can be brainstormed and refactored later on, without modifying the sdk's API. 
Let's focus on the API in this PR.